### PR TITLE
Table favorites

### PR DESCRIPTION
**Pull request by jschuler**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/5020

https://patternfly-react-pr-5213.surge.sh/components/table#favoritable

### CODE REVIEW

**Comment by mcarrano** [comment]
This looks great @jschuler . There's just one small styling issue that I wonder if we can fix. When the favorites star in the header is not selected to sort, it should be a slightly darker gray per the design. Here's what the original design looked like:

Screen Shot 2020-12-04 at 9 38 04 AM

and here's the implementation:

Screen Shot 2020-12-04 at 9 39 08 AM

Right now it feels like the column is disabled. @mcoker do we need a CSS update for that? @mceledonia can you tell us what gray value we should be using there? I don't see it specified in the design. Thanks.

**Comment by mcoker** [comment]
@mcarrano I want to say that was something we changed while implementing the PR, but I can make it darker.

**Comment by mceledonia** [comment]
Yeah it should be our secondary text/global color #6A6E73 in order for it to pass contrast requirements.

**Review by mcarrano — CHANGES_REQUESTED** [review]
@jschuler the styling looks right now. In looking at this again, there's just one thing I was hoping you can change. When I first click the star in the header to sort by favorities, the initial sort is to put the favorited items at the end. I need to click again to move them to the top. That's exactly opposite of what I expect. If a user sorts by favorities, it's most likely because they want to pop all of the favorited items to the top of the table so they can take some action on them. Is is possible to just turn that around?

File: packages/react-table/src/components/Table/examples/Table.md [hunk 1]
```diff
@@ -1812,6 +1816,171 @@ class EditableRowsTable extends React.Component {
...
+          onChange={this.toggleFavsSort}
+          aria-label="toggle select all checkbox"
+          id="toggle-select-all"
+          name="toggle-select-all"
+        />
+        <Table
+          // using this prop enables the favorites column
+          onFavorite={this.onFavorite}
+          // if the onSort prop is detected, favorites can be sorted
+          // if you want to exclude favorites from sorting you can use this prop with a value of `false`
```
Comment (tlabaj) [inline packages/react-table/src/components/Table/examples/Table.md:1966]: These comments are great.  I wonder if it makes sense to add them to the prop docs?

File: packages/react-table/src/components/TableComposable/Td.tsx [hunk 2]
```diff
@@ -63,6 +64,16 @@ export interface TdProps extends BaseCellProps, Omit<React.HTMLProps<HTMLTableDa
...
   };
+  favorites?: {
+    /** Whether the corresponding row is favorited */
+    isFavorited: boolean;
+    /** Callback on clicking the favorites button */
+    onFavorite?: OnFavorite;
+    /** The row index */
+    rowIndex?: number;
+    /** Additional props forwarded to the FavoritesCell */
+    props?: any;
```
Comment (tlabaj) [inline packages/react-table/src/components/TableComposable/Td.tsx:75]: DO you think we should make the name more specific here?

File: packages/react-table/src/components/Table/examples/Table.md (hunk 1 above)
Comment (jschuler) [reply packages/react-table/src/components/Table/examples/Table.md:1966]: yes, I added a description for the Favorites example that explains how to enable favoriting, and in the props I added
```
/**
   * Enables favorites column
   * Callback triggered when a row is favorited/unfavorited
   */
  onFavorite?: OnFavorite;
  /** Along with the onSort prop, enables favorites sorting, defaults to true */
  canSortFavorites?: boolean;
```

**Review by tlabaj — COMMENTED** [review]
Looks good. Can we add some cypress interaction test as well.

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks for making that change @jschuler .

**Review by kmcfaul — APPROVED** [review]
lgtm code wise!

File: packages/react-table/src/components/TableComposable/Td.tsx (hunk 2 above)
Comment (jschuler) [reply packages/react-table/src/components/TableComposable/Td.tsx:75]: To use it you need to wrap it in the `favorites` object which gives it context, e.g.
```
favorites: {
  props: { 'aria-label': 'favs' }
}
```

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by mcoker — APPROVED** [review]
LGTM! 🤩

**Pull request merged**